### PR TITLE
niri: Init at unstable-2023-10-31

### LINKED ISSUE #262804: Package request: niri
**Project description**

 A scrollable-tiling Wayland compositor. 

**Metadata**

* homepage URL: https://github.com/YaLTeR/niri
* source URL: https://github.com/YaLTeR/niri
* license: GPL-3.0
* platforms: linux, ?

### PR DESCRIPTION
**Pull request by IogaMaster**
## Description of changes

Closes #262804 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by IogaMaster** [comment]
This package does build on darwin, let me know if it need's to be linux only.
idk darwin use cases

File: pkgs/applications/window-managers/niri/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    wayland
+    libinput
+    mesa # libgbm
+  ];
+
+  meta = with lib; {
+    description = "A scrollable-tiling Wayland compositor";
+    homepage = "https://github.com/YaLTeR/niri";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ ];
```
Comment (Lurkki14) [inline pkgs/applications/window-managers/niri/default.nix:51]: You need to add yourself as a maintainer https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/applications/window-managers/niri/default.nix (hunk 1 above)
Comment (IogaMaster) [reply pkgs/applications/window-managers/niri/default.nix:51]: Okay

**Comment by Lurkki14** [comment]
> This package does build on darwin, let me know if it need's to be linux only. idk darwin use cases

I'm not too knowledgable about this but https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/applications/window-managers/dwl/default.nix#L97 seems sensible

**Review by Lurkki14 — APPROVED** [review]

**Comment by IogaMaster** [comment]
Result of `nixpkgs-review pr 264517` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>niri</li>
  </ul>
</details>

**Comment by KiaraGrouwstra** [comment]
for the record, this still has me run into the (apparently currently nix-specific) run-time issue reported at https://github.com/YaLTeR/niri/pull/77#issuecomment-1798540387.

**Comment by twoneis** [comment]
@KiaraGrouwstra Could you try adding libglvnd to the buildInputs (and parameters) and these rust flags?
```
RUSTFLAGS = map (a: "-C link-arg=${a}") [
    "-Wl,--push-state,--no-as-needed"
    "-lEGL"
    "-lwayland-client"
    "-Wl,--pop-state"
  ];
```
The rust flags are copied from the cosmic-comp nixpkg, I'm not sure if all of them are really necessary. Please let me know if this fixes the issue for you.

**Review by dit7ya — COMMENTED** [review]
There is a release upstream, can you please update to that? And it would also be nice to not check in the cargo lock file if possible.

**Comment by IogaMaster** [comment]
> There is a release upstream, can you please update to that? And it would also be nice to not check in the cargo lock file if possible.

Can do, the lock file is required because of git deps. nix-init did this for me.

File: pkgs/applications/window-managers/niri/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+, libxkbcommon
+, pipewire
+, seatd
+, udev
+, wayland
+, libinput
+, mesa
+}:
+
+rustPlatform.buildRustPackage rec {
```
Comment (oluceps) [inline pkgs/applications/window-managers/niri/default.nix:14]: `rec` here could be removed.

**Comment by KiaraGrouwstra** [comment]
upstream commit they had close the outstanding issue there: https://github.com/YaLTeR/niri/commit/236abd9d9da14d614f9f36f1469ff90f441ba167

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -7643,6 +7643,12 @@
     githubId = 88038050;
     name = "Souvik Sen";
   };
+  iogamaster = {
+    email = "iogamastercode+nixpkgs@gmail.com";
+    name = "IogaMaster";
+    github = "iogamaster";
+    githubId = 67164465;
+  };
```
Comment (wineee) [inline maintainers/maintainer-list.nix:7651]: please make a *separate* commit with the message `maintainers: add iogamaster`

**Review by wineee — CHANGES_REQUESTED** [review]

**Pull request closed without merging**